mesa/visualization/mpl_space_drawing.py

```diff
@@ -323,62 +316,79 @@ def draw_hex_grid(
-        mesh = PatchCollection(
-            patches, edgecolor="k", facecolor=(1, 1, 1, 0), linestyle="dotted", lw=1
-        )
-        return mesh
+    # Calculate proper bounds that account for the full hexagon width and height
+    x_max = space.width * x_spacing + (space.height % 2) * (x_spacing / 2)
+    y_max = space.height * y_spacing
+
+    # Add padding that accounts for the hexagon points
+    x_padding = (
+        size * np.sqrt(3) / 2
+    )  # Distance from center to rightmost point of hexagon
+    y_padding = size  # Distance from center to topmost point of hexagon
+
+    # Plot limits to perfectly contain the hexagonal grid
+    # Determined through physical testing.
+    ax.set_xlim(-2 * x_padding, x_max + x_padding)
+    ax.set_ylim(-2 * y_padding, y_max + y_padding)
+
+    def setup_hexmesh(width, height):
+        """Helper function for creating the hexmesh with unique edges."""
+        edges = []
+        size = 1.0
+        x_spacing = np.sqrt(3) * size
+        y_spacing = 1.5 * size
+
+        def get_hex_vertices(
+            center_x: float, center_y: float
+        ) -> list[tuple[float, float]]:
+            """Get vertices for a hexagon centered at (center_x, center_y)."""
+            vertices = [
+                (center_x, center_y + size),  # top
+                (center_x + size * np.sqrt(3) / 2, center_y + size / 2),  # top right
+                (center_x + size * np.sqrt(3) / 2, center_y - size / 2),  # bottom right
+                (center_x, center_y - size),  # bottom
+                (center_x - size * np.sqrt(3) / 2, center_y - size / 2),  # bottom left
+                (center_x - size * np.sqrt(3) / 2, center_y + size / 2),  # top left
+            ]
+            return vertices
+
+        # Generate edges for each hexagon
+        for row, col in itertools.product(range(height), range(width)):
+            # Calculate center position for each hexagon with offset for even rows
+            x = col * x_spacing + (row % 2) * (x_spacing / 2)
+            y = row * y_spacing
+
+            vertices = get_hex_vertices(x, y)
+
+            # Edge logic, connecting each vertex to the next
+            for v1, v2 in pairwise([*vertices, vertices[0]]):
+                # Sort vertices to ensure consistent edge representation and avoid duplicates.
+                edge = tuple(sorted([tuple(np.round(v1, 6)), tuple(np.round(v2, 6))]))
+                if edge not in edges:
+                    edges.append(edge)
+
+        # Return LineCollection for hexmesh
+        return LineCollection(edges, linestyle=":", color="black", linewidth=1, alpha=1)
 
     if draw_grid:
-        # add grid
-        ax.add_collection(
-            setup_hexmesh(
-                space.width,
-                space.height,
-            )
-        )
+        ax.add_collection(setup_hexmesh(space.width, space.height))
+
+    # Set aspect ratio to 'equal' to ensure hexagons appear regular
+    ax.set_aspect("equal")
```

Comment:
Please leave this out. Users can do this in their post_process function. Adding it here renders this function inconsistent with the others.